sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/CheckpointStore.cs

```diff
@@ -78,12 +79,38 @@ public abstract class CheckpointStore
         /// <param name="sequenceNumber">An optional sequence number to associate with the checkpoint, intended as informational metadata.  The <paramref name="offset" /> will be used for positioning when events are read.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal a request to cancel the operation.</param>
         ///
-        public abstract Task UpdateCheckpointAsync(string fullyQualifiedNamespace,
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public virtual Task UpdateCheckpointAsync(string fullyQualifiedNamespace,
```

Comment:
Couldn't think of any way around this breaking change

Reply:
Was thinking about this more, and we should probably reverse the call flow here.   The hidden method should call the new public method passing no replication segment or client identifier.  

The new method would throw by default.  Both would still need to be `virtual` rather than abstract.